engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/StorageSystemSnapshotStrategy.java

```diff
@@ -92,7 +90,34 @@
 
     @Override
     public SnapshotInfo backupSnapshot(SnapshotInfo snapshotInfo) {
-        return snapshotInfo;
+
```

Comment:
Please add a ``Preconditions.checkArgument(snapshotInfo != null, "<explaination of the precondition>")`` to protect against potential NPEs.